docs/rl-algorithms/ppo_dna.md

```diff
@@ -0,0 +1,101 @@
+# Proximal Policy Gradient with Dual Network Architecture (PPO-DNA)
+
+## Overview
+
+PPO-DNA is a more sample efficient variant of PPO, based on using separate optimizers and hyperparameters for the actor (policy) and critic (value) networks.
+
+Original paper: 
+
+* [DNA: Proximal Policy Optimization with a Dual Network Architecture](https://arxiv.org/abs/2206.10027)
+
+## Implemented Variants
+
+
+| Variants Implemented      | Description |
+| ----------- | ----------- |
+| :material-github: [`ppo_dna_atari_envpool.py`](https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/ppo_dna_atari_envpool.py), :material-file-document: [docs](/rl-algorithms/ppo_dna/#ppo_dna_atari_envpoolpy) | Uses the blazing fast Envpool Atari vectorized environment. |
+
+Below are our single-file implementations of PPO-DNA:
+
+## `ppo_dna_atari_envpool.py`
+
+The [ppo_dna_atari_envpool.py](https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/ppo_dna_atari_envpool.py) has the following features:
+
+* Uses the blazing fast [Envpool](https://github.com/sail-sg/envpool) vectorized environment.
+* For Atari games. It uses convolutional layers and common atari-based pre-processing techniques.
+* Works with the Atari's pixel `Box` observation space of shape `(210, 160, 3)`
+* Works with the `Discrete` action space
+
+???+ warning
+
+    Note that `ppo_dna_atari_envpool.py` does not work in Windows :fontawesome-brands-windows: and MacOs :fontawesome-brands-apple:. See envpool's built wheels here: [https://pypi.org/project/envpool/#files](https://pypi.org/project/envpool/#files)
+
+
+### Usage
+
+```bash
+poetry install -E envpool
+python cleanrl/ppo_dna_atari_envpool.py --help
+python cleanrl/ppo_dna_atari_envpool.py --env-id Breakout-v5
+```
+
+### Explanation of the logged metrics
+
+See [related docs](/rl-algorithms/ppo/#explanation-of-the-logged-metrics) for `ppo.py`.
+
+### Implementation details
+
+[ppo_dna_atari_envpool.py](https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/ppo_dna_atari_envpool.py) uses a customized `RecordEpisodeStatistics` to work with envpool but has the same other implementation details as `ppo_atari.py` (see [related docs](/rl-algorithms/ppo/#implementation-details_1)).
+
+Note that the original DNA implementation uses the `StickyAction` environment pre-processing wrapper (see (Machado et al., 2018)[^1]), but we did not implement it in [ppo_dna_atari_envpool.py](https://github.com/vwxyzjn/cleanrl/blob/master/cleanrl/ppo_dna_atari_envpool.py) because envpool for now does not support `StickyAction`.
+
+
+### Experiment results
+
+Below are the average episodic returns for `ppo_dna_atari_envpool.py` compared to `ppo_atari_envpool.py`.
+
+
+| Environment      | `ppo_dna_atari_envpool.py` | `ppo_atari_envpool.py` |
+| ----------- | ----------- | ----------- | 
+| BattleZone-v5 (40M steps) | 74000 ± 15300 | 28700 ± 6300 |
+| BeamRider-v5 (10M steps) | 5200 ± 900 | 1900 ± 530 |
+| Breakout-v5 (10M steps) | 319 ± 63 | 349 ± 42 |
```

Comment:
I suggest being consistent and always using 50M steps. It's ok to compare the `ppo_atari_envpool.py` that only has 10M steps. If you think comparing 50M steps is warranted, please start a `ppo_atari_envpool_50M` wandb project in the `openrlbenchmark` entity and run `ppo_atari_envpool.py` in 50M steps there.

Reply:
Sure, though I'm using rented cloud GPU's so re-running has a direct cost to me.

Also, I think there's already a couple of layers of evidence for PPO-DNA without running any more experiments:
1. Primarily, I was interested in reproducing the paper results on the five Atari envs shown in Fig 6 in the paper. The paper used 50M steps for their experiments, so that's what I did as well. My results are not exactly the same, but in the same ballpark (one env performs better, two are comparable, and two are worse).
2. Secondarily, I wanted to do an apples-to-apples comparison against the baseline `ppo_atari_envpool.py` implementation, because that way the environment settings and other implementation details wouldn't distort the results. The CleanRL documentation has learning curves for Breakout, Pong, and BeamRider for 10M steps, so that's what I ran (plus I noticed some Tennis runs in the W&B project). I attempted to run the comparisons for 50M steps as well, but the baseline runs crashed (due to my cloud account running dry), so I thought 40M steps is already more than enough.

Therefore, I'm not sure what the benefit from re-running `ppo_atari_envpool.py` for 50M steps would be?

With regards to hyperparameters, they are otherwise the same, except for total_timesteps.